File: front/lib/data/models/community_model/dating_guide.dart
class DatingGuide {
  String? dgNo;
  String? title;
  String? contents;
  String? thumb;
  String? category;
  String? userNo;
  int? heart;
  DateTime? regDate;

  String? userName;
  String? userProfile;

  @override
  String toString() {
    return 'DatingGuide{dgNo: $dgNo, title: $title, contents: $contents, thumb: $thumb, category: $category, userNo: $userNo, heart: $heart, regDate: $regDate, userName: $userName, userProfile: $userProfile}';
  }

  DatingGuide.fromJson(Map<String, dynamic> json)
      : dgNo = json['dgNo'],
        title = json['title'],
        contents = json['contents'],
        thumb = json['thumb'],
        category = json['cateName'],
        userNo = json['userNo'],
        heart = json['heart'],
        regDate = DateTime.parse(json['regDate']),
        userName = json['userName'],
        userProfile = json['imageUrl'];
}

const Map<String, int> datingGuideCate = {
  '연애 준비': 1,
  '연애 첫걸음': 2,
  '오래가는 연애': 3,
};

File: back/src/main/java/com/pingo/entity/community/DatingGuide.java
package com.pingo.entity.community;

import com.pingo.exception.BusinessException;
import com.pingo.exception.ExceptionCode;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.ToString;

import java.time.LocalDateTime;
import java.util.UUID;

@Getter
@ToString
@AllArgsConstructor
@NoArgsConstructor
public class DatingGuide {
    private String dgNo;
    private String title;
    private String contents;
    private String thumb;
    private int category;
    private String userNo;
    private int heart;
    private LocalDateTime regDate;

    public void createDgNo() {
        if(this.dgNo == null) {
            this.dgNo = "DG" + createNo();
        }else {
            throw new BusinessException(ExceptionCode.DUPLICATE_DATING_GUIDE_NO);
        }
    }

    public String createThumbName() {
        if(this.thumb == null) {
            return "GI" + createNo();
        }else {
            throw new BusinessException(ExceptionCode.DUPLICATE_GUIDE_IMAGE_NO);
        }
    }
    private String createNo() {
        return UUID.randomUUID().toString().substring(0, 8);
    }
    public void insertThumb(String imageUrl) {
        this.thumb = imageUrl;
    }

}
